### PR TITLE
fix: add lottery re-run functionality

### PR DESCRIPTION
**Pull request by ludtkemorgan**
This PR addresses #4246 

- [x] Addresses the issue in full
- [ ] Addresses only certain aspects of the issue

## Description

Re-run of the lottery needs an extra step of resetting the application lottery positions so that it is starting from scratch.

NOTE: this is pointing to the history log PR because there was a bit of refactor that is not yet in main

## How Can This Be Tested/Reviewed?

This can be tested on the UI. Go to a closed listing and run the lottery. Check the order of the applications in that run. Click the re-run lottery button and re-export the results. The order should be different. Also notice the "re-run" in the history log sidebar

## Author Checklist:

- [ ] Added QA notes to the issue with applicable URLs
- [ ] Reviewed in a desktop view
- [ ] Reviewed in a mobile view
- [ ] Reviewed considering accessibility
- [ ] Added tests covering the changes
- [ ] Made corresponding changes to the documentation
- [ ] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

### CODE REVIEW

**Comment by mcgarrye** [comment]
Just want to confirm a few things. 
1. An admin can re-run a lottery at any time? Or should their be protections against allowing them to re-run when a lottery has been released/published/expired?
2. Is there a follow-up ticket for errors or do we want that logic to go in with the re-run?

**Comment by emilyjablonski** [comment]
Is this rebased off of latest main after the re-run changes went in?

**Comment by emilyjablonski** [comment]
Is this rebased off off latest main after the re-run changes went in?

**Comment by ludtkemorgan** [comment]
> Is this rebased off off latest main after the re-run changes went in?

@emilyjablonski this is pointing towards your history log PR since there is some refactoring happening there. I won't merge this until the history log PR is merged

**Comment by ludtkemorgan** [comment]
> Just want to confirm a few things.
> 
> 1. An admin can re-run a lottery at any time? Or should their be protections against allowing them to re-run when a lottery has been released/published/expired?
> 2. Is there a follow-up ticket for errors or do we want that logic to go in with the re-run?

@mcgarrye 
1. A lottery can be re-run at any time. When it runs the lottery results are automatically retracted
2. Good question. I think it might require a different ticket since a failure could happen at any time and I don't know how the front-end should handle it

**Review by mcgarrye — APPROVED** [review]
Looks good!

**Comment by mcgarrye** [comment]
> > Just want to confirm a few things.
> > 
> > 1. An admin can re-run a lottery at any time? Or should their be protections against allowing them to re-run when a lottery has been released/published/expired?
> > 2. Is there a follow-up ticket for errors or do we want that logic to go in with the re-run?
> 
> @mcgarrye
> 
> 1. A lottery can be re-run at any time. When it runs the lottery results are automatically retracted
> 2. Good question. I think it might require a different ticket since a failure could happen at any time and I don't know how the front-end should handle it

Noting: Re-running anytime is only supported through the API right now, the frontend does not give admins that ability.

**Pull request merged**
